recipes/bdwgc/all/conanfile.py

```diff
@@ -56,88 +56,113 @@ class BdwGcConan(ConanFile):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        if tools.Version(self.version) <= "8.0.6":
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        if Version(self.version) < "8.2.0":
             del self.options.throw_bad_alloc_library
         if not self.options.cplusplus:
-            del self.settings.compiler.libcxx
-            del self.settings.compiler.cppstd
+            try:
+                del self.settings.compiler.libcxx
+            except Exception:
+                pass
+            try:
+                del self.settings.compiler.cppstd
+            except Exception:
+                pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
         if self.settings.os == "Windows":
-            self.requires("libatomic_ops/7.6.10")
+            self.requires("libatomic_ops/7.6.14")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
 
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
+    def generate(self):
+        tc = CMakeToolchain(self)
         for option, _ in self._autotools_options_defaults:
-            self._cmake.definitions["enable_{}".format(option)] = self.options.get_safe(option)
-        self._cmake.definitions["disable_gc_debug"] = not self.options.gc_debug
-        self._cmake.definitions["disable_handle_fork"] = not self.options.handle_fork
-        self._cmake.definitions["install_headers"] = True
-        self._cmake.definitions["build_tests"] = False
-        self._cmake.verbose = True
-        self._cmake.parallel = False
-        self._cmake.configure()
-        return self._cmake
-
-    def _patch_sources(self):
-        for patch in self.conan_data["patches"][self.version]:
-            tools.patch(**patch)
+            if option == "cord":
+                tc.variables["build_cord"] = self.options.get_safe(option)
+            elif option == "cplusplus":
+                tc.cache_variables["enable_cplusplus"] = str(self.options.get_safe(option))
+            else:
+                tc.variables["enable_{}".format(option)] = self.options.get_safe(option)
+        tc.variables["disable_gc_debug"] = not self.options.gc_debug
+        tc.variables["disable_handle_fork"] = not self.options.handle_fork
+        tc.variables["install_headers"] = True
```

Comment:
https://github.com/ivmai/bdwgc/blob/213e4b3b3ff54851034239476d4ec845e10b4770/CMakeLists.txt#L57

```suggestion
        tc.variables["install_headers"] = True
        tc.variables["enable_docs"] = False
```

Reply:
Done